src/module/generate-routes.ts

```diff
@@ -53,3 +53,28 @@ export const generateRoutes = async (
 
   return metadata;
 };
+
+const exactly = (input: SwaggerConfigRelatedToRoutes): SwaggerConfigRelatedToRoutes => {
+  // Validate the config values first
+  if (
+    input.noImplicitAdditionalProperties === undefined ||
+    input.noImplicitAdditionalProperties === 'throw-on-extras' ||
+    input.noImplicitAdditionalProperties === 'silently-remove-extras'
+  ) {
+    // then it's good to go
+  } else {
+    throw new Error(`noImplicitAdditionalProperties is set to an invalid value. See https://github.com/lukeautry/tsoa/blob/master/src/config.ts for available options.`);
+  }
+
+  // Make an exact copy that doesn't have other properties
+  const recordOfProps: Record<keyof SwaggerConfigRelatedToRoutes, 'right side does not matter'> = {
+    noImplicitAdditionalProperties: 'right side does not matter',
+  };
+
+  const exactObj: SwaggerConfigRelatedToRoutes = {};
+  Object.keys(recordOfProps).forEach((key) => {
+    const strictKey = key as keyof SwaggerConfigRelatedToRoutes;
+    exactObj[strictKey] = input[strictKey];
+  });
```

Comment:
I know this is very "meta" but I wanted to explain why I'm stripping the config object. The reason why I'm doing this is because it is possible that people could pass in more values since the CLI and JSON are not typesafe. So this method prunes the config object to only what's allowed. This is necessary since we plop the object into the handlebars template and it shows up as an object literal which is subject to what TypeScript calls ["excess property checks."](https://www.typescriptlang.org/docs/handbook/interfaces.html#excess-property-checks)